tuning/data/data_processors.py

```diff
@@ -383,11 +380,128 @@ def _execute_data_handlers(
+
+                raise NotImplementedError(
+                    "Partial splits for streaming IterableDataset are not supported yet!"
+                )
+            if isinstance(loaded_dataset, IterableDatasetDict):
+                # We can try the same logic but only for the "train" split key
+                missing = [k for k in use_existing_splits if k not in loaded_dataset]
+                if missing:
+                    raise ValueError(
+                        f"Expected splits {missing} missing in IterableDatasetDict"
+                    )
+                base_ds = loaded_dataset["train"]
+                if train_split == 1.0 and validation_split == 0.0:
+                    return IterableDatasetDict({"train": base_ds})
+                if validation_split == 1.0 and train_split == 0.0:
+                    return IterableDatasetDict({"validation": base_ds})
+
+                raise NotImplementedError(
+                    "Partial splits for streaming IterableDatasetDict are not supported yet."
+                )
+
+            raise TypeError(
+                f"Expected streaming dataset type but got {type(loaded_dataset)}"
+            )
+
+        # Non-streaming Dataset or DatasetDict
+        if isinstance(loaded_dataset, DatasetDict):
+            # Use the train split inside DatasetDict
+            missing = [k for k in use_existing_splits if k not in loaded_dataset]
+            if missing:
+                raise ValueError(f"Expected splits {missing} missing in DatasetDict")
+
+            # Using "train" split as default
+            base_ds = loaded_dataset["train"]
+            # Support for [0,1] split ranges
+            if train_split == 1.0 and validation_split == 0.0:
+                return DatasetDict({"train": base_ds})
+            if validation_split == 1.0 and train_split == 0.0:
+                return DatasetDict({"validation": base_ds})
+
+            split_ds = base_ds.train_test_split(
+                test_size=validation_split,
+                train_size=train_split,
+                shuffle=True,
+                seed=seed,
+            )
+
+            # After train-validation split and previously exisiting splits are returned!
+            new_ds = DatasetDict()
+            new_ds["train"] = split_ds["train"]
+            new_ds["validation"] = split_ds["test"]
+            return new_ds
+
+        if isinstance(loaded_dataset, Dataset):
+            # Directly split the Dataset
+            base_ds = loaded_dataset
+            split_ds = base_ds.train_test_split(
+                test_size=validation_split,
+                train_size=train_split,
+                shuffle=True,
+                seed=seed,
+            )
+
+            return DatasetDict(
+                {
+                    "train": split_ds["train"],
+                    "validation": split_ds["test"],
+                }
+            )
+
+        raise TypeError(
```

Comment:
why do we need this check here? is there a reason you expect the types to mismatch?

Reply:
this was to ensure we catch if we receive something other than the following formats!
maybe due to some incomplete load or something!
more like safety catch!
I can remove this if required but it seems it can be good to have..
like just in case you receive None type from load_dataset if it fails or anything other than the required Dataset 
types!